go/vt/vtgate/evalengine/fn_time.go

```diff
@@ -1452,6 +1459,12 @@ func dateTimeUnixTimestamp(env *ExpressionEnv, date eval) evalNumeric {
 	}
 
 	ts := dt.dt.ToStdTime(env.now)
+	lowerBoundTimestamp := time.Date(1970, 01, 01, 0, 0, 0, 0, time.UTC)
+	upperBoundTimestamp := time.Date(3001, 1, 19, 3, 14, 7, 99999, time.UTC)
+	if ts.Before(lowerBoundTimestamp) || ts.After(upperBoundTimestamp) {
+		return newEvalInt64(0)
+	}
```

Comment:
Same here, I think testing against unix time stamps is much simpler here.

Reply:
done.